hail/python/hailtop/utils/utils.py

```diff
@@ -431,6 +431,14 @@ async def run_if_changed(changed, f, *args, **kwargs):
             await changed.wait()
 
 
+async def run_if_changed_idempotent(changed, f, *args, **kwargs):
```

Comment:
I spend a while trying to figure out this function, so just to make sure I understand correctly:

The difference here is that we clear the event flag after running the function instead of before. So the "idempotent" here refers to `f` being idempotent? As in, `f` might/will set the event flag, but running it twice in a row produces the same result as running it once, so we don't want to run it again if the event flag gets set from running `f` the first time?

Reply:
I didn't want the behavior if the event is set in the middle of while we're querying BigQuery to have the query repeat itself because the clear happened before the function completed.